app/views/layouts/application.html.erb

```diff
@@ -1,15 +1,51 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <title>MembersOnly</title>
+    <title>Members Only</title>
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
 
     <%= stylesheet_link_tag 'application', media: 'all', 'data-turbolinks-track': 'reload' %>
+    <%= stylesheet_link_tag "https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" %>
     <%= javascript_pack_tag 'application', 'data-turbolinks-track': 'reload' %>
   </head>
 
   <body>
+    <% if flash[:notice] %>
+      <div class="alert alert-primary py-1" role="alert">
+        <p class="notice"><%= notice %></p>
+      </div>
+    <% end %>
+    <% if flash[:alert] %>
+      <div class="alert alert-danger py-0" role="alert">
+        <p class="alert"><%= alert %></p>
+      </div>
+    <% end %>
+
+    <nav class="navbar navbar-expand-lg navbar-light bg-light">
+      <div class = "w-50">
+        <a class="navbar-brand text-info" href="#">
+          <%= link_to "Members Only", root_path %>
+        </a>
+      </div>
+      <div class = "d-flex w-50 justify-content-end">
+        <% if user_signed_in? %>
+          <div>
+            <%= link_to current_user.name, edit_user_registration_path, class: "btn btn-info" %>
+            <%= link_to "Logout", destroy_user_session_path, method: :delete, class: "btn btn-info"%>
+          </div>
+        <% else %>
+          <div>
+            <button class="btn btn-outline-success mx-2" type="submit">
+              <%= link_to "Sign In", new_user_session_path %>       
+            </button>
+            <button class="btn btn-outline-success mx-2" type="submit">
+              <%= link_to "Sign Up", new_user_registration_path %>
+            </button>
+          </div>
+        <% end %>
```

Comment:
View-related logic can be encapsulated in View helpers (app/helpers/*_helper.rb)